### ISSUE TITLE
PRETEND_VERSION_FOR_DIST_NAME does not work with setup.cfg

### ISSUE THREAD

**Issue by tomprince** [pypa/setuptools-scm#597]
It appears that `SETUPTOOLS_SCM_PRETEND_VERSION_FOR_DIST_NAME` does not work when the dist name is in `setup.cfg`, since `Distribution.finalize_options` (which is where `infer_version` is hooked) is called before `Distribution`parse_config_files`.

**Comment by RonnyPfannschmidt** [comment]
@jaraco i beleive tis is yet another setuptools issue thats hard to follow u with

@tomprince would it be reasonable enough for you if it still worked if use_scm_version was set in setup.py while this is being resolved

**Comment by jaraco** [comment]
I commented on this in discord.

**Comment by tomprince** [comment]
I've tested this against master, and it appears to work as expected. Thanks!

As an aside, I was a bit confused looking at the releases sidebar on https://github.com/pypa/setuptools_scm. It appears that there are newer releases on pypi (6.0.0 and 6.0.1) than are shown there. I guess if you've had any *github* releases, it will continue to show them, even if you stop using that feature. I'm not sure there is a good solution, though.

**Comment by RonnyPfannschmidt** [comment]
@tomprince the releases where live and got yanked due to regressions that didn't show on our ci

i did have a facepalm friday with that
